src/query/models/tags.go

```diff
@@ -386,6 +386,29 @@ func (t Tags) AddOrUpdateTag(tag Tag) Tags {
 	return t.AddTag(tag)
 }
 
+// AddDedupedTags is used to add a list of tags with unique names
+// and maintain sorted order.
+func (t Tags) AddDedupedTags(tags []Tag) Tags {
```

Comment:
Maybe we could rename this `AddTagsIfNotExists(...)`?  

Only issue with the current name is that it's not 100% clear what happens when a de-dupe occurs (i.e. does it clobber or leave the original).

Reply:
Yeah that's much better